File: check.js
const fs = require('fs');
const prices = JSON.parse(fs.readFileSync('scripts/supplier-prices.json', 'utf8'));
const products = JSON.parse(fs.readFileSync('scripts/israel-judaica-products.json', 'utf8'));
const scrapedSkus = new Set(products.map(p => p.sku));
const newSkus = Object.keys(prices).filter(sku => !scrapedSkus.has(sku));

console.log(SKUs in price file: );
console.log(SKUs in scraped data: );
console.log(NEW SKUs (not in scraped): \n);
newSkus.slice(0, 20).forEach((sku, i) => console.log(${i+1}. : ₪));

File: check.py
import json

prices = json.load(open('scripts/supplier-prices.json'))
products = json.load(open('scripts/israel-judaica-products.json'))
scraped_skus = set(p['sku'] for p in products)
new_skus = [sku for sku in prices.keys() if sku not in scraped_skus]

print(f'SKUs in price file: {len(prices)}')
print(f'SKUs in scraped data: {len(scraped_skus)}')
print(f'NEW SKUs (not in scraped): {len(new_skus)}')
print()
for i, sku in enumerate(new_skus[:20], 1):
    print(f'{i}. {sku}: {prices[sku]}')
